pms3003.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * pms3003.c
+ *
+ *  Created on: Jan 31, 2017
+ *      Author: Tom
+ */
+
+// Standard includes
+#include <stdarg.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+// Driverlib includes
+#include "hw_types.h"
+#include "hw_memmap.h"
+#include "prcm.h"
+#include "pin.h"
+#include "uart.h"
+#include "rom.h"
+#include "rom_map.h"
+
+#if defined(USE_FREERTOS) || defined(USE_TI_RTOS)
+#include "osi.h"
+#endif
+
+#include "pms3003.h"
+
+#define UartGetChar()       MAP_UARTCharGet(PMS)
+#define BUFFER_LENGTH       24
+#define PM01_HIGH           4
+#define PM01_LOW            5
+#define PM2_5_HIGH          6
+#define PM2_5_LOW           7
+#define PM10_HIGH           8
+#define PM10_LOW            9
+
+//*****************************************************************************
+//
+//! Initialization
+//!
+//! This function
+//!        1. Configures the UART to be used.
+//!
+//! \return none
+//
+//*****************************************************************************
+void
+InitPMS()
+{
+  MAP_UARTConfigSetExpClk(PMS,MAP_PRCMPeripheralClockGet(PMS_PERIPH),
+                    UART1_BAUD_RATE, (UART_CONFIG_WLEN_8 | UART_CONFIG_STOP_ONE |
+                     UART_CONFIG_PAR_NONE));
+}
+
+
+//*****************************************************************************
+//
+//! FillBuffer
+//!
+//!    @brief  Fill buffer with data from particulate matter sensor
+//!
+//!     @param  *buf is the buffer to fill
+//!
+//!     @return filled buffer
+//!
+//!
+//
+//*****************************************************************************
+long*
+FillBuffer(long *buf)
+{
+
+    // TODO: Wait state / error checking / RTOS functionality (osi sleep)
+    int i;
+    for(i=0;i<BUFFER_LENGTH;i++){
+        if(!MAP_UARTBusy(PMS)){

```

Comment:
This is not correct. According to the CC3200 SDK, the UARTBusy function "Returns true if the UART is transmitting or false if all
transmissions are complete."

We are not transmitting any bytes over UART, we are simply receiving them from the PMS3003 sensor.